.github/workflows/test_and_lint.yml

```diff
@@ -0,0 +1,78 @@
+name: test and lint
+
+on:
+  push:
+    branches: [ github-actions ]
+  pull_request:
+    branches: [ github-actions ]
+
+jobs:
+
+  build-and-test:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        node-version: [10.x]
+
+    steps:
+    - name: install dependencies on ubuntu
+      if: startsWith(matrix.os,'ubuntu')
+      run: |
+        sudo apt install -y make gcc pkg-config build-essential libx11-dev libxkbfile-dev 
+    
+    - name: install dependencies on windows
+      if: startsWith(matrix.os,'windows')
+      run: |
+        npm --add-python-to-path install --global --production windows-build-tools
+      shell: powershell 
+    
+    - uses: actions/checkout@v2
+    
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+    
+    - name: Cache node_modules with yarn
+      uses: actions/cache@v2
+      id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+    
+    - run: yarn --frozen-lockfile
+    
+    - run: yarn test
+      env:
+        CI: true
+        
+  code-lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out Git repository
+      uses: actions/checkout@v2
+
+    - name: Use Node.js 10.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 10
+
+    # ESLint and Prettier must be in `package.json`
+    - run: yarn --frozen-lockfile
+
+    - uses: samuelmeuli/lint-action@v1
```

Comment:
This part I wasn't sure. Do we just drop if the pipeline fail ? Or we try to auto-fix if possible then auto recommit ? 
This action does that, it supports auto-fix and repush/recommit.

Reply:
How sure are we that the auto-fix will work 100% of the time? Instead of this, I think I would prefer turning-on autofix in the IDE (but would not force it on everyone)